crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -68,9 +71,44 @@ impl VisibleEntities {
     }
 }
 
+#[derive(Debug, Default, Clone)]
+pub struct EntityMeshRelationships {
+    entities_with_mesh: HashMap<Handle<Mesh>, SmallVec<[Entity; 1]>>,
+    mesh_for_entity: HashMap<Entity, Handle<Mesh>>,
+}
+
+impl EntityMeshRelationships {
+    /// Register the passed `entity` has having the passed `mesh_handle`.
+    ///
+    /// Note that this list _could_ have duplicates if an entity is assigned a new mesh and
+    /// then re-assigned the old mesh. This case should be rare so, for now, we'll risk
+    /// duplicating `Aabb` cloning+assigning.
+    fn register(&mut self, entity: Entity, mesh_handle: &Handle<Mesh>) {
+        self.entities_with_mesh
+            .entry(mesh_handle.clone_weak())
+            .or_default()
+            .push(entity);
+        self.mesh_for_entity
+            .insert(entity, mesh_handle.clone_weak());
+    }
+
+    /// Deregisters the relationship between an `Entity` and `Mesh`. Used so [`update_bounds`] can
+    /// track which relationships are still active so `Aabb`s are updated correctly.
+    fn deregister(&mut self, entity: Entity) {
```

Comment:
Yeah. Sucks that the runtime is proportional to the number of entities with a `Handle<Mesh>` that gets updated or removed, so that removal of large quantities, or updating a single mesh that is shared across many entities. I don't think it's possible to get around it though. I'd suggest documenting that in the `update_bounds` system. It's important that users are aware of cost and benefits of various functionalities.

Reply:
Documented in f9df58adecf5b03d1d6c6845eeda69b219ac65bd - lemme know if that makes enough sense!